File: Day_4/python/day4_patterns.py
"""Day 4 pattern programs using nested loops."""


def square_star_pattern(n: int) -> None:
    for _ in range(n):
        for _ in range(n):
            print("*", end=" ")
        print()


def right_triangle_star_pattern(n: int) -> None:
    for i in range(1, n + 1):
        for _ in range(i):
            print("*", end=" ")
        print()


def number_triangle_pattern(n: int) -> None:
    for i in range(1, n + 1):
        for j in range(1, i + 1):
            print(j, end=" ")
        print()


def reverse_triangle_star_pattern(n: int) -> None:
    for i in range(n, 0, -1):
        for _ in range(i):
            print("*", end=" ")
        print()


def square_number_pattern(n: int) -> None:
    for i in range(1, n + 1):
        for _ in range(1, n + 1):
            print(i, end=" ")
        print()


print("Day 4 Pattern Problems")
print("1. Square star pattern")
print("2. Right triangle star pattern")
print("3. Number triangle pattern")
print("4. Reverse triangle star pattern")
print("5. Square number pattern")

choice = int(input("Choose a pattern (1-5): "))
n = int(input("Enter n: "))

if choice == 1:
    square_star_pattern(n)
elif choice == 2:
    right_triangle_star_pattern(n)
elif choice == 3:
    number_triangle_pattern(n)
elif choice == 4:
    reverse_triangle_star_pattern(n)
elif choice == 5:
    square_number_pattern(n)
else:
    print("Invalid choice")

File: Day_4/java/Day4Patterns.java
import java.util.Scanner;

public class Day4Patterns {
    static void squareStarPattern(int n) {
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }

    static void rightTriangleStarPattern(int n) {
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= i; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }

    static void numberTrianglePattern(int n) {
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= i; j++) {
                System.out.print(j + " ");
            }
            System.out.println();
        }
    }

    static void reverseTriangleStarPattern(int n) {
        for (int i = n; i >= 1; i--) {
            for (int j = 1; j <= i; j++) {
                System.out.print("* ");
            }
            System.out.println();
        }
    }

    static void squareNumberPattern(int n) {
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                System.out.print(i + " ");
            }
            System.out.println();
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Day 4 Pattern Problems");
        System.out.println("1. Square star pattern");
        System.out.println("2. Right triangle star pattern");
        System.out.println("3. Number triangle pattern");
        System.out.println("4. Reverse triangle star pattern");
        System.out.println("5. Square number pattern");
        System.out.print("Choose a pattern (1-5): ");
        int choice = sc.nextInt();
        System.out.print("Enter n: ");
        int n = sc.nextInt();

        if (choice == 1) {
            squareStarPattern(n);
        } else if (choice == 2) {
            rightTriangleStarPattern(n);
        } else if (choice == 3) {
            numberTrianglePattern(n);
        } else if (choice == 4) {
            reverseTriangleStarPattern(n);
        } else if (choice == 5) {
            squareNumberPattern(n);
        } else {
            System.out.println("Invalid choice");
        }

        sc.close();
    }
}

